docker/Dockerfile

```diff
@@ -0,0 +1,104 @@
+# build stage for compiling odamex
+FROM ubuntu:20.04 AS build
+WORKDIR /build
+ENV DEBIAN_FRONTEND=noninteractive
+RUN true \
+    && apt-get update -qq \
+    && apt-get install -y --no-install-recommends tzdata\
+    && apt-get install -y -qq --no-install-recommends \
+        ca-certificates \
+        git \
+        gcc g++ \
+        cmake \
+        make \
+        deutex \
+        libssl-dev \
+        libsdl2-dev libsdl2-mixer-dev \
+        libpng-dev libpng++-dev \
+        libcurl4-openssl-dev libcurlpp-dev \
+        patch \
+        > /dev/null
+
+# So we can use bash arrays (default /bin/sh doesn't support this)
+SHELL ["/bin/bash","-c"]
+
+ARG REPO_URL
+ARG REPO_TAG
+
+# Clone the repository
+RUN true \
+    && test -n "$REPO_URL" && test -n "$REPO_TAG" \
+    && git clone --depth 1 --branch "$REPO_TAG" "$REPO_URL" odamex
+
+WORKDIR /build/odamex
+
+# Apply Manual Patches (make sure they are UTF-8 encoded)
+RUN mkdir /patches
+COPY docker-files/patches /patches
+RUN true \
+    && shopt -s nullglob \
+    && for p in /patches/*.patch; do patch -p1 < $p; done
+
+# Update submodules
+RUN true \
+    && git submodule init \
+    && git submodule update
+
+# Setup Build
+RUN true \
+    && mkdir build
+
+# Build the Wad
+# WORKDIR /build/odamex/wad
+# RUN true \
+#     && /usr/games/deutex -rgb 0 255 255 -doom2 bootstrap -build wadinfo.txt ../odamex.wad
+
+# Build Odamex
+WORKDIR /build/odamex/build
+RUN true \
+    && cmake -W no-dev \
+        -D CMAKE_BUILD_TYPE=Release \
+        -D CMAKE_CXX_COMPILER=g++ \
+        -D CMAKE_C_COMPILER=gcc \
+        .. \
```

Comment:
If you add `-D BUILD_CLIENT=0 -D BUILD_LAUNCHER=0`, it will skip building some of their dependencies that are unneeded for the server.